go/arrow/array/table.go

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package array
+
+import (
+	"sync/atomic"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/internal/debug"
+)
+
+// Chunked manages a collection of primitives arrays as one logical large array.
+type Chunked struct {
+	chunks []Interface
+
+	refCount int64
+
+	length int
+	nulls  int
+	dtype  arrow.DataType
+}
+
+// NewChunked returns a new chunked array from the slice of arrays.
+//
+// NewChunked panics if the chunks do not have the same data type.
+func NewChunked(dtype arrow.DataType, chunks []Interface) *Chunked {
+	arr := &Chunked{
+		chunks:   make([]Interface, len(chunks)),
+		refCount: 1,
+		dtype:    dtype,
+	}
+	for i, chunk := range chunks {
+		if chunk.DataType() != dtype {
+			panic("arrow/array: mismatch data type")
+		}
+		chunk.Retain()
+		arr.chunks[i] = chunk
+		arr.length += chunk.Len()
```

Comment:
@sbinet - Could this cause issues since length on Chunked is only an int and this is taking the total length of the chunks? Seems like length should be an int64.

Reply:
yeah... could you file a JIRA issue?